tests/unit/v1/test_rate_limiter.py

```diff
@@ -19,7 +19,7 @@
 
 # Pick a point in time as the center of our universe for this test run.
 # It is okay for this to update every time the tests are run.
-fake_now = datetime.datetime.utcnow()
+fake_now = datetime.datetime.now(tz=datetime.timezone.utc)
```

Comment:
This might be outside the scope of the PR you want to make here, but you may want to consider using [`time-machine`](https://github.com/adamchainz/time-machine) or, if that's infeasible for some reason, [`freezegun`](https://github.com/spulec/freezegun) instead of the current mocking approach.

The way it works now, you need to make sure that the thing you are patching returns something with the correct "awareness" (i.e. returns a naïve datetime or aware datetime), but with one of those frameworks, the awareness of the result is dictated by the `datetime` methods that you called, as it would be if you weren't mocking.

Reply:
Yeah I think this may be an improvement to be done as a separate PR

Reply:
Switched to `freezegun` in 5f9f04f1eae89956cbcfe3c08e41638ac976e80f